### PR TITLE
Add mrack.conf which contains default enfironment

### PR DESCRIPTION
**Pull request by Tiboris**
Add mrack.conf from which we load defaults

Add mrack.conf which contains default enfironment
configuration in ini format for the click options
- `--config=provisioning-config from mrack.conf`
- `--db=mrackdb from mrack.conf`

Load default for options from the mrack.conf file from:
1. use `$(pwd)/mrack.conf` or other way the `./mrack.conf` - actual directory
2. or use user home directory configuration - `~/.mrack/mrack.conf`
3. alternatively directory `/etc/mrack/mrack.conf` as safe fallback.

Add example `mrack.conf` and `provisioning-config.yaml`
to the `data/` directory in repository.

Update setup.py to install required files from `data/` dir.

Signed-off-by: Tibor Dudlák <tdudlak@redhat.com>

### CODE REVIEW

**Comment by pvoborni** [comment]
The code seems to expect that the config will be placed in a system directory. This can be a possibility but it doesn't fulfill the original use case. The config file should be loaded from the directory where mrack is executed - in IdM CI terminology - the TWD. If it is not present then it will be ignored. Alternatively mrack can try other locations like `~/.config/mrack` and the system one as a last option. In either case it should work without the file present.

File: src/mrack/data/provisioning-config.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
+aws:
```
Comment (pvoborni) [inline src/mrack/data/provisioning-config.yaml:1]: I don't think we should have a default provisioning config with so many values specific to certain environment. Basically it should not contain anything which is not universal.

File: src/mrack/run.py [hunk 2]
```diff
@@ -43,6 +46,58 @@
...
+def get_config_file_path():
+    """Locate mrack.conf file."""
+    try:
+        return os.path.join(get_mrack_dir(), "mrack.conf")
+    except BrokenInstallation:
+        logger.warning("Can not load mrack.conf using Python package defaults.")
+        return os.path.join(os.path.dirname(__file__), "data/mrack.conf")
+
+
+def get_default_file_path(filename):
```
Comment (pvoborni) [inline src/mrack/run.py:73]: Why not to work with key names directly? What is the benefit of using key names similar to some default values when you will get an error anyway if you specify a different filename when calling the method.

I.e. the method seems to me too complex without  a clear benefit.

File: src/mrack/data/provisioning-config.yaml (hunk 1 above)
Comment (netoarmando) [reply src/mrack/data/provisioning-config.yaml:1]: I'm inclined to agree, does that mean we need "mostly" 1 image, 1 distro, 1 flavor etc?
Some projects have a default config file with lots of comment blocks, explaining every parameter with one (or none) default value enabled. As a user I would appreciate that.

Comment (pvoborni) [reply src/mrack/data/provisioning-config.yaml:1]: +1, commented examples could be good.

Comment (Tiboris) [reply src/mrack/data/provisioning-config.yaml:1]: @netoarmando Will do so

**Review by pvoborni — COMMENTED** [review]
Originally I wanted to approve this with minor comments. But the more I tried it and e.g. found, that writing metadata is still also inconvenient. I started to think more about other use cases and the implementation. And in the end tried a different approach  in https://github.com/neoave/mrack/pull/55 due to the following reasons:
- the get_file_path_from_config method is too overloaded - it does 3 separate things - constructing list of possible paths,  loading the config, finding a value
- is inefficient - it is doing the file operations again and again for each key to load
- hard to extend - it is focusing only on paths, not thinking in a future that we want to store other configuration. E.g. debug level, provisioning strategy or default provisioner.
- due to it doing a lot of stuff it is also hard to read and understand quickly

**Pull request closed without merging**